src/pages/surveys/[id]/questions/[order].page.tsx

```diff
@@ -19,21 +22,29 @@ const Question: NextPage = () => {
   const surveyId = router.query.id as string;
 
   const {
-    store: { currentSurvey },
+    store: { currentSurvey, surveyResponses },
+    dispatchAction,
   } = useContext(StoreContext);
 
-  const [currentQuestion, setCurrentQuestion] =
-    useState<SurveyQuestionInterface>({} as SurveyQuestionInterface);
+  const currentQuestion = useMemo(
+    () =>
+      currentSurvey?.questions.find(
+        ({ displayOrder }) => displayOrder === questionOrder
+      ),
+    []
+  ) as SurveyQuestionInterface;
+
   const [isLoading, setIsLoading] = useState(true);
-  const [answers, setAnswers] = useState(null);
+  const [responses, setResponses] = useState<SurveyResponse | null>(null);
```

Comment:
```suggestion
  const [response, setResponse] = useState<SurveyResponse | null>(null);
```

You might also need to change line 42 `const response = surveyResponses.find(` so avoid naming conflict 🤔 
Maybe using `currentResponse` in one of the case would do ;-)